data/completion/complete.sh

```diff
@@ -0,0 +1,73 @@
+# -*- bash -*-
+
+# _complete_from_snap serialises the tab completion request and sends it off to
+# the appropriate 'snap run --command=complete', and de-serialises the response
+# into the usual tab completion result.
+_complete_from_snap() {
+    {
+        read -a opts
+        # opts is expected to be a series of compopt options
+        if [[ ${#opts[@]} -gt 0 ]]; then
+            if [[ "${opts[0]}" == "cannot" ]]; then
+                # older snap-execs sent errors over stdout :-(
+                return 1
+            fi
+
+            for i in "${opts[@]}"; do
+                if ! [[ "$i" =~ ^[a-z]+$ ]]; then
+                    # non-alphanumeric option; something awry
+                    return 2
+                fi
+            done
+        fi
+
+        read bounced
+        case "$bounced" in
+            ""|"alias"|"export"|"job"|"variable")
+                # OK
+                ;;
+            *)
+                # unrecognised bounce
+                return 2
+                ;;
+        esac
+
+        read sep
+        if [ "$sep" ]; then
+            # non-blank separator? madness!
+            return 2
+        fi
+        local oldIFS="$IFS"
```

Comment:
Hmm, it might not have been IFS related and instead terminal related. It was an unfortunate situation. There is likely work to be done in this area when a reproducer is found/a bug comes in.